lib/libxdp/tests/test_link_dispatch.c

```diff
@@ -0,0 +1,111 @@
+ /* SPDX-License-Identifier: GPL-2.0 */
+  
+#define _GNU_SOURCE
+
+#include <errno.h>
+#include <linux/err.h>
+#include <net/if.h>
+#include <pthread.h>
+#include <stdbool.h>
+#include <stdlib.h>
+#include <string.h>
+#include <sys/resource.h>
+#include <unistd.h>
+
+#include "test_utils.h"
+
+#include <xdp/libxdp.h>
+#include <bpf/libbpf.h>
+
+
+static void usage(char *progname)
+{
+	fprintf(stderr, "Usage: %s <ifname>\n", progname);
+	exit(EXIT_FAILURE);
+}
+
+static int check_link_detach(int ifindex) {
+    struct bpf_object *obj_prog = NULL;
+    struct bpf_program *prog;
+    struct xdp_multiprog *mp;
+    struct bpf_link *link;
+    int ret;
+
+    if (!ifindex) {
+        return -EINVAL;
+    }
+
+     obj_prog = bpf_object__open("xdp_pass.o");
+     if (!obj_prog) {
+        ret = -errno;
+        goto out;
+     }
+
+     prog = bpf_object__find_program_by_name(obj_prog, "xdp_pass");
+     if (!prog) {
+        ret = -errno;
+        goto out;
+     }
+
+
+    ret = bpf_object__load(obj_prog);
+    if (ret) {
+        ret = -errno;
+        fprintf(stderr, "Couldn't load object: %s\n", strerror(-ret));
+        goto out;
+    }
+
+    link = bpf_program__attach_xdp(prog, ifindex);
+    if (!link) {
+        ret = -errno;
```

Comment:
Instead of returning the errno here, I think we should exit with the special return code that makes the test framework mark the test as skipped (249 - add a define for it, though). That way the CI won't fail on this test on older kernels (which I suspect it does now).

We could have a separate detection step first, but that would basically end up doing the same that this function does up to this point, so I think it's easier to just have a direct return of SKIPPED_TEST here :)